spec.md

```diff
@@ -8,121 +8,160 @@ title: Semantic Specification
+                                       ↑
+                                       ↑
+                           (Span G "Follows" Span F)
 
 ~~~
 
 ~~~
-Temporal relationships for spans from a single trace
+Temporal relationships for all of the above
+
 
 ––|–––––––|–––––––|–––––––|–––––––|–––––––|–––––––|–––––––|–> time
 
  [Span A···················································]
    [Span B··············································]
       [Span D··········································]
     [Span C········································]
-         [Span E···········]   [Span F········]
-                                [Span G······]
-                                [Span H··]
-                                        [Span I·]
+         [Span E·······]        [Span F··] [Span G··] [Span H··]
 ~~~
 
 ## Traces
 
-A **Trace** represents the potentially distributed, potentially concurrent data/execution path in a (potentially distributed, potentially concurrent) system. A Trace can be thought of as a tree of Spans. (See the ASCII diagrams above)
+A **Trace** represents the potentially distributed, potentially concurrent data/execution path in a (potentially distributed, potentially concurrent) system. A Trace can be thought of as a directed acyclic graph (DAG) of Spans.
 
 ## Spans
 
-A **Span** represents a logical unit of work in the system that has a start time and a duration. Spans may be nested and ordered to model parent-child and causal relationships. Each span has an **operation name**, a presumably human-readable string which concisely names the work done by the span (e.g., an RPC method name, a function name, or the name of a subtask within a larger computation).
+A **Span** represents a logical unit of work in the system that has a start time and a duration. Spans may be nested and ordered to model causal relationships. Each span has an **operation name**, a presumably human-readable string which concisely represents the work done by the span (e.g., an RPC method name, a function name, or the name of a subtask or stage within a larger computation).
+
+<span id="references"></span>
+
+### Causal Span References
+
+A Span may reference zero or more Spans that are causally related. OpenTracing presently defines two types of references: `BelongsTo` and `Follows`. **Both reference types specifically model direct causal relationships between a descendant span and an ancestor span.** In the future, OpenTracing may also support reference types for spans with non-causal relationships (e.g., Spans that are batched together, Spans that are stuck in the same queue, etc).
+
+**`BelongsTo` references:** A descendant Span may "BelongTo" an ancestor Span. In a "BelongsTo" reference, the ancestor Span depends on the descendant Span in some fashion (often—but not always—the ancestor Span cannot finish until the descendant Span finishes). We sometimes refer to the "BelongsTo" ancestor as the "parent" and the descendant as the "child".
```

Comment:
The good and bad news here is that all of these various viewpoints are valid and defensible, and none is a perfect fit.

For the "dependent" case, I am going to go with `ChildOf`. Rationale:
- I think it's important for the term to feel "transitive" in the sense of a "transitive verb": these are directed edges we are decorating, and it must be clear which side is which. @yurishkuro this argues against `Parent(...)` where I think it's potentially confusing whether the span one is starting is a parent or a child. `ChildOf` doesn't have that problem.
- Whether due to brainwashing or not, it is conventional in the distributed tracing universe and being unsurprising is a big plus.
- After sleeping on it, I'd argue that `BelongsTo` implies containment more than causal dependency... also feels jargony per the next bullet
- There are more precise terms (`DependencyOf`, for instance) but I think they are too jargony and probably too verbose as well


For the "independent" case:
- @dkuebric makes the point that both reference types could be considered "children"... I see and recognize that point, but from a drill-down / critical path standpoint, I'd argue that the "dependent" case is more parent-child like than the independent case which is more like a sibling relationship, esp for pipelines.
  - Also, as a parent of an 19-month-old, "Independent Child" is just too much of an oxymoron for my current life stage :)
- `Predecessor` has the same problem as `Parent` from a directed-edge standpoint
- `SuccessorTo` feels to jargony
- `HappensAfter` is ok with me... I wish that it explicitly ruled out the dependent cases, though, as it sounds like `HappensAfter` is a superset of `ChildOf` when it is not intended to be. It may also imply that the new span starts after the ancestor *finishes*, which is not necessarily the case, of course.
- `Follows` is also ok with me, thought it would be nice to match the `<Noun><Preposition>` form of `ChildOf` for consistency. It also has the problem about not ruling out parent-to-child dependencies.
- `SpawnedBy` or `DueTo` might do a better job expressing the independent nature of the newly started span... though they fall into the jargony bucket.

I'll try to make an actual decision and send out an update later today.